src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/Table.cs

```diff
@@ -96,13 +100,29 @@ public override bool CheckTypes(CheckTypesContext context, TexlNode[] args, DTyp
                     }
                 }
 
-                Contracts.Assert(rowType.IsRecord);
+                Contracts.Assert(resultType.IsRecord);
             }
 
-            returnType = rowType.ToTable();
+            returnType = resultType.ToTable();
 
             return isValid;
         }
+
+        public override void CheckSemantics(TexlBinding binding, TexlNode[] args, DType[] argTypes, IErrorContainer errors)
+        {
+            base.CheckSemantics(binding, args, argTypes, errors);
+
+            for (var i = 0; i < argTypes.Length; i++)
+            {
+                var ads = argTypes[i].AssociatedDataSources?.FirstOrDefault();
+
+                if (ads is IExternalDataSource tDsInfo && tDsInfo is IExternalTabularDataSource)
```

Comment:
>ads is IExternalDataSource tDsInfo && tDsInfo is IExternalTabularDataSource [](http://example.com/codeflow?start=20&length=75)

This can be simplified as `if (ads is IExternalTabularDataSource)` - IExternalTabularDataSource inherits from IExternalDataSource